src/main/java/eu/xenit/gradle/enterprise/conventions/extensions/repository/RepositoryHandlerExtensions.java

```diff
@@ -3,59 +3,125 @@
-    private final Project project;
+    private final PropertyReader propertyReader;
 
     protected static final Action<? super MavenArtifactRepository> EMPTY_ACTION = repository -> {
     };
 
     @Inject
-    public RepositoryHandlerExtensions(RepositoryHandler repositoryHandler, Project project) {
+    public RepositoryHandlerExtensions(RepositoryHandler repositoryHandler, PropertyReader propertyReader) {
         this.repositoryHandler = repositoryHandler;
-        this.project = project;
+        this.propertyReader = propertyReader;
+    }
+
+    private Action<? super PasswordCredentials> configureRepoCredentials(String prefix) {
+        String usernameProp = prefix+".username";
+        String passwordProp = prefix+".password";
+
+        if(propertyReader.hasProperty(usernameProp) && propertyReader.hasProperty(passwordProp)) {
+            return passwordCredentials -> {
+                passwordCredentials.setUsername(propertyReader.property(usernameProp).toString());
+                passwordCredentials.setPassword(propertyReader.property(passwordProp).toString());
+            };
+        }
+        return passwordCredentials -> {};
+    }
+
+    @PublicApi
+    public MavenArtifactRepository xenit() {
+        return xenit(EMPTY_ACTION);
+    }
+
+    @PublicApi
+    public MavenArtifactRepository xenit(Action<? super MavenArtifactRepository> action) {
+        return repositoryHandler.maven(repository -> {
+            repository.setName("Xenit");
+            repository.setUrl(StringConstants.XENIT_REPO_URL);
+            repository.credentials(configureRepoCredentials("eu.xenit.repo"));
+            repository.mavenContent(contentDescriptor -> {
+                contentDescriptor.releasesOnly();
+            });
+            action.execute(repository);
+        });
+    }
+
+    @PublicApi
+    public MavenArtifactRepository xenitSnapshots() {
+        return xenitSnapshots(EMPTY_ACTION);
+    }
+
+    @PublicApi
+    public MavenArtifactRepository xenitSnapshots(Action<? super MavenArtifactRepository> action) {
+        return repositoryHandler.maven(repository -> {
+            repository.setName("XenitSnapshots");
+            repository.setUrl(StringConstants.XENIT_REPO_URL);
+            repository.credentials(configureRepoCredentials("eu.xenit.repo"));
+            repository.mavenContent(contentDescriptor -> {
+                contentDescriptor.snapshotsOnly();
+            });
+            action.execute(repository);
+        });
     }
 
     @PublicApi
+    @Deprecated(since = "0.3.0", forRemoval = true)
     public MavenArtifactRepository xenitPrivate() {
         return xenitPrivate(EMPTY_ACTION);
     }
 
     @PublicApi
+    @Deprecated(since = "0.3.0", forRemoval = true)
     public MavenArtifactRepository xenitPrivate(Action<? super MavenArtifactRepository> action) {
+        LOGGER.warn("Using the xenitPrivate() repository is deprecated, as the artifact repository will be shutdown.");
```

Comment:
should we mention that you probably want to use `xenit()` instead ?

Reply:
It's not a drop-in replacement, but I can mention it.